org.eclipse.languageserver/src/org/eclipse/languageserver/operations/completion/LSContentAssistProcessor.java

```diff
@@ -46,34 +36,15 @@ public LSContentAssistProcessor() {
 	@Override
 	public ICompletionProposal[] computeCompletionProposals(ITextViewer viewer, int offset) {
 		ICompletionProposal[] res = new ICompletionProposal[0];
-		IEditorPart activeEditor = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage().getActiveEditor();
-		if (!(activeEditor instanceof AbstractTextEditor)) {
-			return new ICompletionProposal[0];
-		}
-		IEditorInput input = activeEditor.getEditorInput();
-		LanguageClientEndpoint languageClient = null;
-		URI fileUri = null;
-		try {
-			if (input instanceof IFileEditorInput) { // TODO, also support non resource file
-				IFile file = ((IFileEditorInput) input).getFile();
-				fileUri = file.getLocation().toFile().toURI();
-				languageClient = LanguageServiceAccessor.getLanguageServer(file, viewer.getDocument(), ServerCapabilities::isCodeActionProvider);
-			} else if (input instanceof IURIEditorInput) {
-				fileUri = ((IURIEditorInput)input).getURI();
-				// TODO server
-			}
-		} catch (Exception ex) {
-			ex.printStackTrace();
-			return res;
-		}
-		
+        final LSPDocumentInfo info = LSPEclipseUtils.getLSPDocumentInfoFor(viewer);
```

Comment:
It seems to me that this remove the ability to have several language servers for a single file and to pick the right one (using the Predicate of `LanguageServerAccessor.getLanguageServer(file, viewer, predicate)`). This feature is important so the predicate should be an argument of `LSPEclipseUtils.getLSPDocumentInFor`